hpx/config/constexpr.hpp

```diff
@@ -9,7 +9,7 @@
 
 #include <hpx/config/defines.hpp>
 
-#if defined(HPX_HAVE_CXX11_CONSTEXPR) && !defined(HPX_MSVC_NVCC)
+#if defined(HPX_HAVE_CXX11_CONSTEXPR) && !defined(HPX_MSVC_NVCC) && !defined(__CUDACC__)
```

Comment:
Please use `__NVCC__` instead of `__CUDACC__` as the latter is also defined by clang.

Reply:
But constexpress work with cuda clang and we might want to use it there or? It seems they just not work on nvidia nvcc.